components/server/src/server.ts

```diff
@@ -140,14 +141,18 @@ export class Server {
         // Install passport
         await this.authenticator.init(app);
 
-        // log context
-        app.use(async (req: express.Request, res: express.Response, next: express.NextFunction) => {
-            try {
-                const userId = req.user ? req.user.id : undefined;
-                runWithLogContext("http", { userId, requestPath: req.path }, () => next());
-            } catch (err) {
-                next(err);
-            }
+        // Use RequestContext for authorization
+        app.use((req: express.Request, res: express.Response, next: express.NextFunction) => {
+            const userId = req.user ? req.user.id : undefined;
+            runWithRequestContext(
+                {
+                    requestKind: "http",
+                    requestMethod: req.path,
+                    signal: new AbortController().signal,
```

Comment:
not sure whether we want to connect it with req.on('aborted', ...)

Reply:
:shrug: I guess it's the same as [here](https://github.com/gitpod-io/gitpod/pull/19023#discussion_r1395509942): We could argue we can safely add it because we don't check it anywhere, yet... :roll_eyes:

Reply:
yeah, let's not do it for now, but later we should reconsider i.e. in regards to timeouts